### PR TITLE
Add tests for 4.16 EC

### PR DESCRIPTION
**Pull request by liangxia**

### CODE REVIEW

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-ec-aws-ipi-proxy-sts-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-ec-aws-ipi-proxy-sts-f14

**Comment by liangxia** [comment]
Working as expected,
```
INFO[2024-02-02T08:16:20Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release:4.16.0-ec.1-x86_64 
```

/pj-rehearse abort

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+    tag: upi-installer
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.18
+releases:
+  latest:
+    prerelease:
+      architecture: amd64
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:66]: Is amd64 not a default architecture?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+      product: ocp
+      version_bounds:
+        lower: 4.16.0-0
```
Comment (JianLi-RH) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:69]: Just curious can we use below settings to narrow down the scope:
 lower: 4.16.0-ec
 upper: 4.16.0-rc

As I know the version order is:
0.nightly < stable < ec < rc

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:69]: What's the benefit of using `4.16.0-ec` compared with `4.16.0-0` ?

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:69]: I think `4.16.0-0` will include nightly if `4-dev-preview` include nightly build, like: 4.16.0-0.nightly-2024-02-02-002725

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:69]: image

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:69]: Have you ever seen nightly build exist in `4-dev-preview` ?

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:69]: I don't know that, so I said `I curious`

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:69]: From what I know, that issue/curious does not happen till now after we have `4-dev-preview`.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml (hunk 1 above)
Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-ec.yaml:66]: We also use multi in this file, so it's better to explicitly state that.

**Comment by liangxia** [comment]
Rebase and resolve conflicts.

**Pull request merged**